tests/utils.py

```diff
@@ -1087,3 +1089,23 @@ def _is_compile_finished() -> bool:
 
     await retry_limited(_is_compile_finished, timeout=10)
     return run
+
+
+def validate_version_numbers_migration_scripts(versions_folder: pathlib.Path) -> None:
+    """
+    Validate whether the names of the database migration scripts in the given directory
+    are compliant with the schema. Migration scripts must have the format vYYYYMMDDN.py
+    """
+    v1_found = False
+    for path in versions_folder.iterdir():
+        file_name = path.name
+        if not file_name.endswith(".py"):
+            continue
+        if file_name == "__init__.py":
+            continue
+        if file_name == "v1.py":
+            v1_found = True
+            continue
+        if not re.fullmatch(r"v([0-9]{8}[0-9]+)\.py", file_name):
```

Comment:
```suggestion
        if not re.fullmatch(r"v([0-9]{8}[0-9])\.py", file_name):
```
I don't think we currently support more than 10 scripts on the same day. Iirc we use numeric sort on them so adding an extra digit would mess up the ordering.

On a less serious note: I wonder how this will break first: when we reach the year 10k, or if we'll have switched to a new calendar before then.